redbot/cogs/reports/reports.py

```diff
@@ -0,0 +1,322 @@
+import logging
+import asyncio
+import discord
+from discord.ext import commands
+from redbot.core import Config, checks, RedContext
+from redbot.core.utils.chat_formatting import pagify, box
+from redbot.core.bot import Red
+from redbot.core.i18n import CogI18n
+
+from .antispam import AntiSpam
+
+_ = CogI18n("Reports", __file__)
+
+log = logging.getLogger("red.reports")
+
+
+class Reports:
+
+    default_guild_settings = {
+        "output_channels": [],
+        "active": False,
+        "next_number": 1,
+        "reports": {}  # Mapping index -> {userid: int, report: str}
+        # index for this starts at one. No user facing report #0
+    }
+
+    # TODO: make this actually configureable later
+    # Intervals should be a list of tuples in the form
+    # (period: seconds, max_frequency: int)
+    # corresponding to the maximum uses per second
+    # see ./antispam.py for more details
+    default_global_settings = {
+        "intervals": []
+    }
+
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.settings = Config.get_conf(
+            self, 78631113035100160, force_registration=True)
+        self.settings.register_guild(**self.default_guild_settings)
+        self.settings.register_global(**self.default_global_settings)
+        self.antispam = {}
+        self.user_cache = []
+
+    @checks.admin_or_permissions(manage_server=True)
+    @commands.guild_only()
+    @commands.group(name="reportset")
+    async def reportset(self, ctx: RedContext):
+        """
+        settings for reports
+        """
+        if ctx.invoked_subcommand is None:
+            return await ctx.send_help()
+
+    @checks.admin_or_permissions(Manage_server=True)
+    @reportset.command(name="output", pass_context=True, no_pm=True)
```

Comment:
`pass_context=True` isn't necessary
`no_pm=True` should be removed and `@commands.guild_only()` should be put on a new line